### DISCUSSION TITLE
What is a sampler and how to choose it here?

### DISCUSSION THREAD

**Discussion by pollow** [lllyasviel/Fooocus#1223] [Q&A] [answered]
I saw many places mentioning "Sampler DPM++ 2M Karras" blah blah. I wonder what does that mean and how to choose it here? Since I didn't see any thing about sampler in the advanced tab. This was mentioned almost everytime when people share their prompt, and even in the Juggernaut XL's description (https://civitai.com/models/133005?modelVersionId=198530).

I'm very new to this and cannot understand some concepts, would really appreciate any help. Thanks.

**Comment by xhoxye** [comment, marked as answer]
Because it's in Developer Debug Mode

![QQ截图20231206211005](https://github.com/lllyasviel/Fooocus/assets/129571231/935e5977-094d-4e28-93a3-541c6b4e7465)

**Comment by vghost01** [comment]
This thread is the first google search result for "fooocus what is a sampler" but it does not actually answer the question of what a sampler is.

**Reply by theoldbar** [reply]
[https://websim.ai/c/I0nMJ7gYtQqdjW7yD](url)

there ya go 😊

And just because someone says this is the best scheduler/ sampler to use it doesn't mean it will be the best for you. each combination alters the image slightly , so the best thing to do ( although its time consuming ) is to test out the various combinations see which one you like best.

for example if i  use a pony model i normally go for eular ancestral + exponential , or if its an sdxl model dpmpp_2m + karras.

its very easy to lose hours trying out the various combinations, i know i've done it lol
